demo/src/main/java/com/orange/ods/demo/ui/components/floatingactionbuttons/ComponentFloatingActionButton.kt

```diff
@@ -0,0 +1,115 @@
+ *  license that can be found in the LICENSE file or at
+ *  https://opensource.org/licenses/MIT.
+ * /
+ */
+
+package com.orange.ods.demo.ui.components.floatingactionbuttons
+
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.FabPosition
+import androidx.compose.material.rememberBottomSheetScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import com.orange.ods.compose.component.OdsComponent
+import com.orange.ods.compose.component.button.OdsExtendedFloatingActionButton
+import com.orange.ods.compose.component.button.OdsFloatingActionButton
+import com.orange.ods.compose.component.chip.OdsChoiceChip
+import com.orange.ods.compose.component.chip.OdsChoiceChipsFlowRow
+import com.orange.ods.compose.component.list.OdsListItem
+import com.orange.ods.compose.component.list.OdsSwitchTrailing
+import com.orange.ods.demo.R
+import com.orange.ods.demo.ui.components.utilities.ComponentCustomizationBottomSheetScaffold
+import com.orange.ods.demo.ui.components.utilities.clickOnElement
+import com.orange.ods.demo.ui.utilities.composable.CodeImplementationColumn
+import com.orange.ods.demo.ui.utilities.composable.FloatingActionButtonTechnicalText
+
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+fun ComponentFloatingActionButton() {
+    val context = LocalContext.current
+    val fabCustomizationState = rememberFabCustomizationState()
+
+    with(fabCustomizationState) {
+        if (size.value == FabCustomizationState.Size.Mini) {
+            resetTextAndWidth()
+        }
+
+        val modifier = Modifier
+            .let {
+                if (isFullScreenWidth) it
+                    .padding(
+                        start = dimensionResource(id = R.dimen.spacing_m),
+                        end = dimensionResource(id = R.dimen.spacing_m),
+                        bottom = 64.dp
+                    )
+                    .fillMaxWidth()
+                else it
+            }
+
+        ComponentCustomizationBottomSheetScaffold(
+            bottomSheetScaffoldState = rememberBottomSheetScaffoldState(),
+            floatingActionButton = {
+                if (hasText) {
+                    OdsExtendedFloatingActionButton(
+                        onClick = {
+                            clickOnElement(context, context.getString(R.string.component_floating_action_button))
+                        },
+                        text = stringResource(id = R.string.component_floating_action_button_add),
+                        icon = painterResource(id = R.drawable.ic_plus),
```

Comment:
Should we use a recipe icon instead ? (I would say that "add" is better but I'm just wondering)